xhr.js

```diff
@@ -0,0 +1,30 @@
+var APP = APP || {};
+(function(app) {
+
+    var API_URLROOT = 'https://api.stackexchange.com/2.2/questions/';
+
+    app.xhr = {};
+
+    app.xhr.fetch = function(ids) {
+        return new Promise(function(resolve, reject) {
+            var req = new XMLHttpRequest();
+            var url = app.xhr.makeApiUrl(ids);
+            req.onreadystatechange = function() {
+                if (req.readyState === 4) {
+                    if (req.status === 200) {
+                        return resolve(JSON.parse(req.responseText));
+                    } else {
+                        return reject('shit dun goofed');
```

Comment:
Don't throw an error, it's not safe inside the callback, but do wrap the `resolve` call in a try catch in case invalid JSON is received (otherwise it'd behave incorrectly. Also, it's probably better to use `.onload` insteado f `.onreadystatechange`